src/tls/connection.c

```diff
@@ -790,10 +791,18 @@ connection_crypto_init (const char *certfile,
   if (ret != GNUTLS_E_SUCCESS)
     errx (EXIT_FAILURE, "gnutls_certificate_allocate_credentials failed: %s", gnutls_strerror (ret));
 
+  /* check if we have a separate key file */
+  keyfile = cockpit_certificate_key_path (certfile);
   if (keyfile)
```

Comment:
Which means that this `if` is weird...

Reply:
Yes, the `if`s can go away once _key_path asserts, and we only need to handle the ENOENT case.